core/src/main/java/jenkins/security/apitoken/ApiTokenStore.java

```diff
@@ -176,18 +176,50 @@ private void addLegacyToken(@Nonnull Secret legacyToken, boolean migrationFromEx
         // 16x8=128bit worth of randomness, using brute-force you need on average 2^127 tries (~10^37)
         byte[] random = new byte[16];
         RANDOM.nextBytes(random);
+        // 32-char in hex
         String secretValue = Util.toHexString(random);
         String tokenTheUserWillUse = HASH_VERSION + secretValue;
         assert tokenTheUserWillUse.length() == 2 + 32;
         
-        String secretValueHashed = this.plainSecretToHashInHex(secretValue);
+        HashedToken token = prepareAndStoreToken(name, secretValue);
+        
+        return new TokenUuidAndPlainValue(token.uuid, tokenTheUserWillUse);
+    }
+    
+    /**
+     * Be careful with this method. Depending on how the tokenPlainValue was stored/sent to this method, 
+     * it could be a good idea to generate a new token randomly and revoke this one.
+     */
+    public synchronized @Nonnull String addFixedNewToken(@Nonnull String name, @Nonnull String tokenPlainValue) {
+        if (tokenPlainValue.length() != 2 + 32) {
```

Comment:
It might be more readable to extract these values into named constants. It reduces the need to refer to API docs or comments when reading the code.